### ISSUE TITLE
Godot 4.0 Beta 11 Crash

### ISSUE THREAD

**Issue by DeadlyEssence01** [OrigamiDev-Pete/TODO_Manager#27]
To Do manager seems to crash Godot beta 11 immediately. Disabling the plugin allowed the editor to start.

I ran verbose on Godot's launch, not sure it'll be of any help:

```
Loaded builtin certs
Creating VMA small objects pool for memory type index 3
Failed to bind socket. Error: 3
Failed to bind socket. Error: 3
EditorSettings: Save OK!
Loading resource: res://addons/Todo_Manager/UI/Dock.tscn
Loading resource: res://addons/Todo_Manager/UI/ColourPicker.tscn
Loading resource: res://addons/Todo_Manager/UI/Pattern.tscn

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.beta11.official (91713ced81792b10fdc9367b7f355738e5d52777)
```

I can include the backtrace if it'll help.

**Comment by cridenour** [comment]
Seeing the same - I filed https://github.com/godotengine/godot/issues/71173 which is tangentially related - more of a "why is this silently hard crashing" vs. why is this now out of date. 

But the full stack trace will help this as well:

```
CrashHandlerException: Program crashed
Engine version: Godot Engine v4.0.beta.custom_build (91713ced81792b10fdc9367b7f355738e5d52777)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[0] std::_Atomic_storage<unsigned int,4>::load (C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\atomic:1001)
[1] std::_Atomic_storage<unsigned int,4>::load (C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\atomic:1001)
[2] SafeNumeric<unsigned int>::get (C:\Users\Chris\Development\godot\godot\core\templates\safe_refcount.h:72)
[3] SafeRefCount::_check_unref_sanity (C:\Users\Chris\Development\godot\godot\core\templates\safe_refcount.h:173)
[4] SafeRefCount::unref (C:\Users\Chris\Development\godot\godot\core\templates\safe_refcount.h:193)
[5] Callable::~Callable (C:\Users\Chris\Development\godot\godot\core\variant\callable.cpp:348)
[6] Callable::`scalar deleting destructor'
[7] Variant::_clear_internal (C:\Users\Chris\Development\godot\godot\core\variant\variant.cpp:1368)
[8] Variant::clear (C:\Users\Chris\Development\godot\godot\core\variant\variant.h:304)
[9] Variant::~Variant (C:\Users\Chris\Development\godot\godot\core\variant\variant.h:778)
[10] Variant::`scalar deleting destructor'
[11] GDScriptFunction::call (C:\Users\Chris\Development\godot\godot\modules\gdscript\gdscript_vm.cpp:3541)
[12] GDScriptInstance::callp (C:\Users\Chris\Development\godot\godot\modules\gdscript\gdscript.cpp:1847)
[13] Object::callp (C:\Users\Chris\Development\godot\godot\core\object\object.cpp:711)
[14] Variant::callp (C:\Users\Chris\Development\godot\godot\core\variant\variant_call.cpp:1161)
[15] GDScriptFunction::call (C:\Users\Chris\Development\godot\godot\modules\gdscript\gdscript_vm.cpp:1583)
[16] GDScriptInstance::callp (C:\Users\Chris\Development\godot\godot\modules\gdscript\gdscript.cpp:1847)
[17] Node::_gdvirtual__ready_call<0> (C:\Users\Chris\Development\godot\godot\scene\main\node.h:242)
[18] Node::_notification (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:156)
[19] Node::_notificationv (C:\Users\Chris\Development\godot\godot\scene\main\node.h:46)
[20] CanvasItem::_notificationv (C:\Users\Chris\Development\godot\godot\scene\main\canvas_item.h:45)
[21] Control::_notificationv (C:\Users\Chris\Development\godot\godot\scene\gui\control.h:47)
[22] Object::notification (C:\Users\Chris\Development\godot\godot\core\object\object.cpp:792)
[23] Node::_propagate_ready (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:189)
[24] Node::_set_tree (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:2595)
[25] Node::_add_child_nocheck (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:1123)
[26] Node::add_child (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:1139)
[27] EditorNode::add_bottom_panel_item (C:\Users\Chris\Development\godot\godot\editor\editor_node.cpp:5343)
[28] EditorPlugin::add_control_to_bottom_panel (C:\Users\Chris\Development\godot\godot\editor\editor_plugin.cpp:426)
[29] call_with_ptr_args_ret_helper<EditorPlugin,Button *,Control *,String const &,0,1> (C:\Users\Chris\Development\godot\godot\core\variant\binder_common.h:324)
[30] call_with_ptr_args_ret<EditorPlugin,Button *,Control *,String const &> (C:\Users\Chris\Development\godot\godot\core\variant\binder_common.h:573)
[31] MethodBindTR<EditorPlugin,Button *,Control *,String const &>::ptrcall (C:\Users\Chris\Development\godot\godot\core\object\method_bind.h:476)
[32] GDScriptFunction::call (C:\Users\Chris\Development\godot\godot\modules\gdscript\gdscript_vm.cpp:1945)
[33] GDScriptInstance::callp (C:\Users\Chris\Development\godot\godot\modules\gdscript\gdscript.cpp:1847)
[34] Node::_gdvirtual__enter_tree_call<0> (C:\Users\Chris\Development\godot\godot\scene\main\node.h:240)
[35] Node::_propagate_enter_tree (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:218)
[36] Node::_set_tree (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:2591)
[37] Node::_add_child_nocheck (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:1123)
[38] Node::add_child (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:1139)
[39] EditorNode::add_editor_plugin (C:\Users\Chris\Development\godot\godot\editor\editor_node.cpp:3357)
[40] EditorNode::set_addon_plugin_enabled (C:\Users\Chris\Development\godot\godot\editor\editor_node.cpp:3490)
[41] EditorNode::_notification (C:\Users\Chris\Development\godot\godot\editor\editor_node.cpp:631)
[42] EditorNode::_notificationv (C:\Users\Chris\Development\godot\godot\editor\editor_node.h:113)
[43] Object::notification (C:\Users\Chris\Development\godot\godot\core\object\object.cpp:792)
[44] Node::_propagate_ready (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:189)
[45] Node::_propagate_ready (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:181)
[46] Node::_set_tree (C:\Users\Chris\Development\godot\godot\scene\main\node.cpp:2595)
[47] SceneTree::initialize (C:\Users\Chris\Development\godot\godot\scene\main\scene_tree.cpp:411)
[48] OS_Windows::run (C:\Users\Chris\Development\godot\godot\platform\windows\os_windows.cpp:1275)
[49] widechar_main (C:\Users\Chris\Development\godot\godot\platform\windows\godot_windows.cpp:181)
[50] _main (C:\Users\Chris\Development\godot\godot\platform\windows\godot_windows.cpp:203)
[51] main (C:\Users\Chris\Development\godot\godot\platform\windows\godot_windows.cpp:217)
[52] WinMain (C:\Users\Chris\Development\godot\godot\platform\windows\godot_windows.cpp:231)
[53] __scrt_common_main_seh (D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[54] <couldn't map PC to fn name>
-- END OF BACKTRACE --
```

**Comment by OrigamiDev-Pete** [comment]
Good catch. There definitely seems to be a regression in GDScript. Thanks @cridenour for raising an issue in the Godot repo.

**Comment by OrigamiDev-Pete** [comment]
I just pushed a hotfix for this. If you're able to clone and test the latest update, let me know how it goes.
See my reply to godotengine/godot#71173 for more details of the bug.

**Comment by OrigamiDev-Pete** [comment]
This should be resolved by godotengine/godot#71228

**Comment by officialduke99** [comment]
I would like to confirm, that this is no longer an issue. Am running Godot 4 Beta 11 and upgraded to plugin version 2.1.1 and it's working.
